zingolib/src/wallet/keys/unified.rs

```diff
@@ -421,31 +400,9 @@ impl WalletCapability {
             None
         };
 
-        let transparent_receiver = if desired_receivers.transparent {
-            let transparent_fvk: AccountPubKey = match self.unified_key_store().try_into() {
-                Ok(viewkey) => viewkey,
-                Err(e) => {
-                    self.addresses_write_lock
-                        .swap(false, atomic::Ordering::Release);
-                    return Err(e.to_string());
-                }
-            };
-
-            match self.generate_transparent_receiver(
-                &transparent_fvk,
-                TransparentKeyScope::EXTERNAL,
-                legacy_key,
-            ) {
-                Ok(t_addr) => Some(t_addr),
-                Err(e) => {
-                    self.addresses_write_lock
-                        .swap(false, atomic::Ordering::Release);
-                    return Err(e.to_string());
-                }
-            }
-        } else {
-            None
-        };
+        let transparent_receiver = self
```

Comment:
why have we removed the desired_receivers check and also the fvk generation so its unified with the logic of sapling and orchard? I did this work on my last PR and its much cleaner and important to prevent bugs with UFVKs without transparent viewing keys

Reply:
Oh lord...I think this may have been lost when I was trying to fix merge conflicts.

Reply:
Is this resolved to your satisfaction @Oscar-Pepper ?

Reply:
Yes, I think on one version it was moved, and when I resolved conflicts it was now in neither location...added back, good catch.

Reply:
@AloeareV has re-added it.